pkg/secrets/secrets.go

```diff
@@ -180,16 +180,19 @@ func SecretMountsWithUIDGID(mountLabel, containerWorkingDir, mountFile, mountPre
 		}
 	}
 
-	// Add FIPS mode secret if /etc/system-fips exists on the host
-	_, err := os.Stat("/etc/system-fips")
-	if err == nil {
-		if err := addFIPSModeSecret(&secretMounts, containerWorkingDir, mountPrefix, mountLabel, uid, gid); err != nil {
-			logrus.Errorf("error adding FIPS mode secret to container: %v", err)
+	// Only add FIPS secret mount if the label FIPS_DISABLE=true is not set in the pod/container
+	if !disableFips {
```

Comment:
Nit: A positive check would simplify the code a bit more.
```golang
if disableFips {
    return secretMounts
}
_, err := os.Stat("/etc/system-fips")
...
```

Reply:
Fixed. Thanks for the pointer :)